packages/editor/src/components/post-preview-button/index.js

```diff
@@ -197,15 +197,18 @@ export default compose( [
 			getCurrentPostAttribute,
 			getAutosaveAttribute,
 			getEditedPostAttribute,
+			getAutosave,
 			isEditedPostSaveable,
 			isEditedPostAutosaveable,
 		} = select( 'core/editor' );
 		const {
 			getPostType,
 		} = select( 'core' );
 		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		const postId = getCurrentPostId();
 		return {
-			postId: getCurrentPostId(),
+			postId,
+			autosave: getAutosave( postId ),
```

Comment:
I'm not actually using this to disable use of the preview button yet. That's because it's possible for a post to not have an autosave and the preview button should still be usable. I think there'll need to be some separate state like 'hasCheckedForAutosave' that is stored, and when that's false the preview button is disabled.